backtesting/backtesting.py

```diff
@@ -998,6 +1005,10 @@ def _process_orders(self):
             # If we don't have enough liquidity to cover for the order, the broker CANCELS it
             if abs(need_size) * adjusted_price_plus_commission > \
                     self.margin_available * self._leverage:
+                warnings.warn(
+                    f'({data.index[self._i]}) broker canceled order {order} due to insufficient margin '
+                    f'(equity={self.equity:.2f}, margin_available={self.margin_available:.2f}).',
+                    category=UserWarning)
```

Comment:
This hunk is legit, I think. :+1: If it can be made to match the prior art in https://github.com/kernc/backtesting.py/commit/5e72e7b270a468ae3747c36678292f6469f48f82 ...